cli/resource_group.py

```diff
@@ -0,0 +1,29 @@
+import logging
+
+from azure.mgmt.resource import ResourceManagementClient
+from msrestazure.azure_cloud import AZURE_PUBLIC_CLOUD
+
+import cli.utils
+
+LOGGER = logging.getLogger(__name__)
+RG_TYPE = 'resource group'
+
+def create_rg_if_not_exists(context):
+    """Create a new resource group."""
+    resource_group = context.obj['resource_group']
+    location = context.obj['location']
+    context.obj['resource_client'] = ResourceManagementClient(
+        credentials=context.obj['aad_credentials'],
+        subscription_id=context.obj['subscription_id'],
+        base_url=AZURE_PUBLIC_CLOUD.endpoints.resource_manager)
+
+    if not resource_group_exists:
+        context.obj['resource_client'].resource_groups.create_or_update(resource_group,
+                                                                        {'location': location})
```

Comment:
Long lines with short endings like this one look weird; maybe if the ending is short its ok to put it all on the same line? Or maybe it can be broken up something like this:
```
context.obj['resource_client']
    .resource_groups.create_or_update(resource_group,
                                      {'location': location})
```
or perhaps
```
context.obj['resource_client']
.resource_groups.create_or_update(resource_group,
                                  {'location': location})
```
but to be honest both of these look weird too. Maybe a slightly-too-long one liner is the way to go.

maybe this?
```
context.obj['resource_client']
    .resource_groups.create_or_update(
        resource_group,
        {'location': location}
    )
```
this one is probably the least pythonic, but its a little cleaner

Reply:
I like the last one haha. I changed it and linted it and it was fine.... I think it's a bit javascripty but its easier to read IMO as well.